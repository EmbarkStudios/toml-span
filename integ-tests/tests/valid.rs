use integ_tests::valid;

valid!(comments);
valid!(evil);

mod arrays {
    use super::valid;

    valid!(empty, "thevoid = [[[[[]]]]]");
    valid!(no_spaces, "ints = [1,2,3]");
    valid!(heterogenous, r#"mixed = [[1, 2], ["a", "b"], [1.1, 2.1]]"#);
    valid!(nested, r#"nest = [["a"], ["b"]]"#);
    valid!(ints_and_floats, "ints-and-floats = [1, 1.1]");
    valid!(
        ints_and_arrays,
        r#"arrays-and-ints =  [1, ["Arrays are not integers."]]"#
    );
    valid!(strings_and_ints, r#"strings-and-ints = ["hi", 42]"#);
    valid!(
        one,
        "[[people]]\nfirst_name = \"Bruce\"\nlast_name = \"Springsteen\""
    );
}

mod tables {
    use super::valid;

    valid!(
        implicit_and_explicit_after,
        r"
[a.b.c]
answer = 42

[a]
better = 43
"
    );
    valid!(
        implicit_and_explicit_before,
        r"
[a]
better = 43

[a.b.c]
answer = 42
"
    );
    valid!(implicit_groups, "[a.b.c]\nanswer = 42");
    valid!(implicit_array, "[[albums.songs]]\nname = \"Glory Days\"");
    valid!(
        array_many,
        r#"
[[people]]
first_name = "Bruce"
last_name = "Springsteen"

[[people]]
first_name = "Eric"
last_name = "Clapton"

[[people]]
first_name = "Bob"
last_name = "Seger"
"#
    );
    valid!(
        nested_arrays,
        r#"
[[albums]]
name = "Born to Run"

  [[albums.songs]]
  name = "Jungleland"

  [[albums.songs]]
  name = "Meeting Across the River"

[[albums]]
name = "Born in the USA"
  
  [[albums.songs]]
  name = "Glory Days"

  [[albums.songs]]
  name = "Dancing in the Dark"
"#
    );
    valid!(sub_empty, "[a]\n[a.b]");
    valid!(table_span, "\n[workspace]\nmembers = true\nother = false\n[workspace.foo]\nbar=3\n\n[workspace.bar]\ninline = { foo = 3 }\n\n");
}

mod numbers {
    use super::valid;

    valid!(integers);
    valid!(floats);
}
