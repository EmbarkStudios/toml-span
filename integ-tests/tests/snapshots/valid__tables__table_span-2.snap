---
source: integ-tests/tests/valid.rs
expression: spans
---
note[integer]: root_workspace_bar_inline_foo
  ┌─ table_span:9:18
  │
9 │ inline = { foo = 3 }
  │                  ^

note[table]: root_workspace_bar_inline
  ┌─ table_span:9:10
  │
9 │ inline = { foo = 3 }
  │          ^^^^^^^^^^^

note[table]: root_workspace_bar
  ┌─ table_span:8:1
  │  
8 │ ╭ [workspace.bar]
9 │ │ inline = { foo = 3 }
  │ ╰────────────────────^

note[integer]: root_workspace_foo_bar
  ┌─ table_span:6:5
  │
6 │ bar=3
  │     ^

note[table]: root_workspace_foo
  ┌─ table_span:5:1
  │  
5 │ ╭ [workspace.foo]
6 │ │ bar=3
  │ ╰─────^

note[bool]: root_workspace_members
  ┌─ table_span:3:11
  │
3 │ members = true
  │           ^^^^

note[bool]: root_workspace_other
  ┌─ table_span:4:9
  │
4 │ other = false
  │         ^^^^^

note[table]: root_workspace
  ┌─ table_span:2:1
  │  
2 │ ╭ [workspace]
3 │ │ members = true
4 │ │ other = false
  │ ╰─────────────^

note[table]: root
   ┌─ table_span:1:1
   │  
 1 │ ╭ 
 2 │ │ [workspace]
 3 │ │ members = true
 4 │ │ other = false
   · │
10 │ │ 
11 │ │ 
   │ ╰^
